python-spec/src/somacore/data.py

```diff
@@ -65,7 +76,48 @@ def read(
     ) -> "ReadIter[pa.Table]":
         """Reads a user-defined slice of data into Arrow tables.
 
-        TODO: Further per-param documentation.
+        :param coords: for each index dimension, which rows to read.
+            Defaults to ``None``, meaning no constraint -- all IDs.
+        :param column_names: the named columns to read and return.
+            Defaults to ``None``, meaning no constraint -- all column names.
+        :param partitions: an optional ``ReadPartitions`` hint to indicate
+            how results should be organized.
+        :param result_order: order of read results.
+            This can be one of 'row-major', 'col-major', or 'auto'.
+        :param value_filter: an optional [value filter] to apply to the results.
```

Comment:
needs an example, no matter how trivial -- or, a statement that the query language is implementation-dependent

this is super-awkward now that this _code_ in the _spec_ repo now appears as on-line help for the _impl_ repo ......

we've had consistent feedback over the last few months on this project "what's an example of a value-filter query?" -- and now we're in a spot where we've made it difficult to satisfy that legitimate user ask ...

Reply:
@thetorpedodog we are on the same page

* didn't have a specific example before
* it should
* but value-filter language is an impl-repo thing and this is a spec-repo PR so it's not ok for us to put _any_ examlpe here

which is unfortunate

all i'm doing is pointing that out

i didn't ask any action from you here & still am not -- just pointing this out as food for thought

the real solution for the "how do i spell a value filter in a query" is going to have to be unanswerable in a docstring (given that this query _code_ is in the _spec_ repo)

the impl stuff will belong in impl-repo examples

Reply:
and ok i guess i _do_ have an ask after all:

let's go ahead say something of the form "value-filter syntax is implementation-dependent -- please see implementation docs for examples"

Reply:
Added. If we *do* want a canonical value-filter syntax then this can be updated.